urllib3/util/retry.py

```diff
@@ -169,10 +184,10 @@ def get_backoff_time(self):
 
         :rtype: float
         """
-        if self._observed_errors <= 1:
+        if len(self.history) <= 1:
             return 0
 
-        backoff_value = self.backoff_factor * (2 ** (self._observed_errors - 1))
+        backoff_value = self.backoff_factor * (2 ** (len(self.history) - 1))
```

Comment:
This change subtly affects the way the backoff works. In particular, redirects will now count towards the backoff count. This means that redirects and retries can interact in a way to dramatically slow things down. Presumably we should count redirects and retries separately from the perspective of backing off.

We may even want to go further here: perhaps each host should handle its backoff separately.